### ISSUE TITLE
Make it possible for users to add information on reports even if the report is marked closed

### ISSUE THREAD

**Issue by evilaliv3** [globaleaks/globaleaks-whistleblowing-software#4293]
### Proposal

This ticket is to propose to make it possible for  users to add information on reports even if the report is marked closed

### Motivation and context

Currently when a report is marked closed no one is enabled to edit the report for adding additional information.

This seems to be very problematic for some relevant use cases where:
- a whistleblower received the final answer from their recipients need to 1) need to ask for support, e.g reaching out to the organization directly 2) add information proving the case was handled incorrectly making appeal
- recipients need to provide some additional update on the report

For this reason many users like @giorgiofraschini are proposing to always make it possible for users to add information on reports even if the report is marked closed

**Comment by evilaliv3** [comment]
I'm personally in favor of this change.

If we really want to preserve the possibility for recipients to mark a report as not modifiable we should eventually evaluate to implement this as a separated function to be used only when strictly needed but I think whistleblowers should always be entitled to add information on their report and this could be very important in risky circumstances.

\cc @larrykind @elbill @javialama @danielvaknine

**Comment by JaviAlama** [comment]
From our point of view and based on experience:

**1-** Recipients CAN have the need to add PRIVATE information (not visible to the whistleblower) to a closed complaint.
**2-** Anonymous whistleblowers, depending on how their case has been closed, if they are not happy with the resolution, they protest through the same mailbox and then… it becomes a chat. That is why in our case, after the resolution of the investigation is signed and uploaded to the case, we always choose to close the case and not allow any further communication.

In summary, we find it interesting to add PRIVATE comments (from the recipients) to closed complaints, but not from the whistleblowers.

Those who are not satisfied with the decision taken, if they have new information, could send a new report. Perhaps it would be interesting to be able to link or connect cases, this would be more useful in these cases.

**Comment by larrykind** [comment]
Hi everyone, 
this our pint of view about the topic:

1. We agree with @JaviAlama about the need for the recipients to comment and discuss internally about a closed report, without communicate it to the whistlelbower

2. Right now the only way to reopen a case comes from the receiver only without any whistleblower input, being anonymous or not.

In any of the cases wistleblower could never get aware of a case reopening, because the whistleblower has previously seen the report marked as closed: this could make the whistleblower trash his 16digit key. In every case the whistleblower could not contribute to a reopened case. How the whistleblower could know the case is reopened? And how the whistleblower could provide further information for a closed case (for example a "rejected" and then "closed" one)?

Our opinion about it n.2 is to add a functionality for the whistleblower:

For closed cases and for a limited period of time there sould be for the whistleblower a button "Ask for reopen the case - (available until ....)" with a very brief text motivation: Once the button is clicked it will be disabled from the whistleblower interface.
From the receiver side this motivation could be treated as a special marked comment "Asked for reopen", and globally treated like a normal report update. 

Thanks for considering it.

**Comment by JaviAlama** [comment]
@evilaliv3 Don't forget to consider the option of being able to link complaints (internally).  This can be very useful.

**Comment by larrykind** [comment]
thanks @evilaliv3 for the proposal.
I agree with your solution. As proposed in today's meeting, It should be nice to make the whistleblower to load a limited number of comments while status is "closed" with the purpose to push the receiver to reopen the case.

Thanks

**Comment by danielvaknine** [comment]
Hello all, we think it's good that recipients can comment on closed cases!

However, regarding the possibility for the whistleblower to comment on closed cases, we don't quite agree. The fact that the whistleblower can't write more is generally liked amongst our users. If the whistleblower is unsatisfied, they can always submit a new report regarding this (which requires a bit more effort than just writing something new in the comments).

Also, we think it might be valuable with an explanation for the whistleblower about why they can't write in the comments. It could be a really short snippet on/above the comment section, like "Report has been closed".

In summary:

1. We agree that recipients should be able to comment on closed cases
2. We _don't_ agree that whistleblowers also should
3. We suggest a clarification to the whistleblower when a report is closed, so they don't see the closed comments as a "bug"

**Comment by giorgiofraschini** [comment]
Hi all! My view on this.
GlobaLeaks exists as a tool to allow interaction between a whistleblower and recipient(s). Which is the reason to close the possibility for the whistleblower to add information, request clarifications, etc about a report?
If the whistleblower finds the report closed without the possibility to keep interacting with the recipient he/she will find other ways  for doing it (creating a new report? reaching out the recipient through different channels?).
I think the label on the status of the report (new/open/closed) should be considered in relation to the status of processing the report and should not be connected with the possibility for the whistleblower to add comments. etc.

Why should a recipient block a whistleblower from doing so? I think recipients should keep the possibility to close comments to the whistleblower but this should not be related to closing a report. i believe many recipients do not know that they are closing comments and select "closed" just to have it "closed" in their reports' list.

**Comment by evilaliv3** [comment]
Thank you.

At the moment to improve current situation with a minor change we:
- have enabled recipients and whistleblowers to comment on closed reports
- have enabled recipients also to upload files.

I think we could rename this ticket as "Evaluate improvements to reports
close state" to continue discuss further improvements
